experimental/generation/generator/packages/library/src/mergeAssets.ts

```diff
@@ -294,148 +292,182 @@ async function mergeRootFile(schemaName: string, oldPath: string, oldFileList: s
-                if (oldListEntitySection.ListBody[i].match(':')) {
-                    const enumEntity = oldListEntitySection.ListBody[i].replace('-', '').replace(':', '').trim()
-                    enumSet.add(enumEntity)
-                    if (enumValueMap.has(enumEntity)) {
-                        resultStatements.push(oldListEntitySection.ListBody[i])
-                    }
-                    let j = i + 1
-                    while (j < oldListEntitySection.ListBody.length) {
-                        if (!oldListEntitySection.ListBody[j].match(':')) {
-                            const enumSyn = oldListEntitySection.ListBody[j].replace('-', '').trim()
-                            if (enumValueMap.has(enumEntity)) {
-                                resultStatements.push(oldListEntitySection.ListBody[j])
-                                keepEnumValue.add(enumSyn)
-                            } else {
-                                deletedEnumValue.add(enumSyn)
-                            }
-                            j++
-                        } else {
-                            i = j - 1
-                            break
-                        }
-                    }
-                }
-            }
+    await writeToFile(oldPath, mergedPath, `${schemaName}.${locale}.lu`, oldFileList, val, feedback)
+}
 
-            // add  new enum entity in the new  lu file 
-            for (const [key, values] of enumValueMap) {
-                if (!enumSet.has(key)) {
-                    resultStatements.push('    - ' + key + ' :')
-                    for (const newStatement of values) {
-                        resultStatements.push(newStatement)
-                    }
-                }
-            }
+const valuePattern = /{@?([^=]*Value)\s*=([^}]*)}/g
+// Use property pattern to check if the utternace is the lu example, notice that value pattern might not always be in the lu example.
+const propertyPattern = /{@?([^=]*Property)\s*=/g
+const commentOutPattern = /^>\s*-/m     
 
-            // update content 
-            const listEntitySectionName = `@ list ${oldListEntitySection.Name} =`
-            const entityLUContent = listEntitySectionName + os.EOL + resultStatements.join(os.EOL)
-            updatedLUResource = oldSectionOp.updateSection(oldListEntitySection.Id, entityLUContent)
+/**
+ * @description: Get the set of deleted utterance patterns.
+ * @param filename Name of the lu file.
+ * @param newFileList List of old file paths.
+ * @param delUtteranceSet Set of deleted utterance patterns.
+ */
+async function getDeletedUtteranceSet(filename: string, oldFileList: string[], delUtteranceSet: Set<string>): Promise<void> {
+    const filePath = oldFileList.filter(file => file.match(filename))[0]
+    const text = await fs.readFile(filePath, 'utf8')
+    const lines = text.split(os.EOL)
+    for (let line of lines) {
+        if (line.match(commentOutPattern) && line.match(propertyPattern)) {
```

Comment:
>&& line.match(propertyPattern) [](start = 42, length = 30)

Remove this test--otherwise people cannot comment out utterances without labels.

Reply:
I think we'd better to use some pattern to constrain which utterance users could comment out, or else, if the user can comment out the following utterance:
@ ml Add() usesFeature AddPhrases
    -@ ml CheeseProperty usesFeature CheeseProperty
        - @ CheeseValue CheeseValue
The lu file might not be correctly trained.

Reply:
If they comment out those parts the file is broken already and we can't prevent them.  There is no pattern we can do for this:
> - cancel
Other than the > - at the start.

---
In reply to: [594935847](https://github.com/microsoft/BotBuilder-Samples/pull/3104#discussion_r594935847) [](ancestors = 594935847)